### DISCUSSION TITLE
How to call a stateNotifierProvider method from a widget dispose method

### DISCUSSION THREAD

**Discussion by SamiAlsubhi** [rrousselGit/riverpod#1610] [Q&A]
I need to call a StateNotifierProvier method from within a dispose method of a widget:
```dart
 @override
  void dispose() {
    super.dispose();
    ref.read(markersProvider.notifier).removeOriginMarker();
  }
```
But I get this error:
```
 EXCEPTION CAUGHT BY WIDGETS LIBRARY
The following assertion was thrown while finalizing the widget tree:
Looking up a deactivated widget's ancestor is unsafe.
At this point the state of the widget's element tree is no longer stable.
To safely refer to a widget's ancestor in its dispose() method, save a reference to the ancestor by
calling dependOnInheritedWidgetOfExactType() in the widget's didChangeDependencies() method.
```
How can I achieve this? 
From what I understand is that this provider is affecting a parent widget and this is not acceptable move inside a child's dispose.
But what could be the solution to this?

**Comment by doanhovy188** [comment]
Move it to **deactive()** override instead of **dispose()**,
it will work!

**Reply by The-Funk** [reply]
@doanhovy188 Been searching for a solution to this problem for a while and your suggestion was just what I needed. Thank you!
